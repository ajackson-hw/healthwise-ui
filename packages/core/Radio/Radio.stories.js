import React from 'react'
import { storiesOf } from '@storybook/react'

import Radio from './index'

const styles = {
  container: {
    fontSize: '24px',
    marginBottom: '16px',
  },
}

storiesOf('core|Form Controls/Radio', module).add(
  'with defaults',
  () => (
    <div>
      <div style={styles.container}>
        <Radio name="test" value="test" />
      </div>
      <div style={styles.container}>
        <Radio name="test 2" value="test" defaultChecked />
      </div>
    </div>
  ),
  {
    info: `Demonstrates default rendering of Radio component`
  }
)
