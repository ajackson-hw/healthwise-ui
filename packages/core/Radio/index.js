import React from 'react'
import PropTypes from 'prop-types'
import styles from './style.css'

class Radio extends React.Component {
  render() {
    let {
      id,
      name,
      value,
      isRequired,
      onClick,
      readonly,
      forPrint,
      ...otherAttributes
    } = this.props

    return (
      <span className={'hw-radio-wrapper ' + styles.radio_wrapper}>
        <input
          type="radio"
          id={id}
          name={name}
          value={value}
          required={isRequired}
          className={'hw-radio ' + styles.radio}
          onClick={onClick}
          disabled={readonly}
          style={forPrint ? { display: 'none' } : null}
          {...otherAttributes}
        />
        <svg
          role="presentation"
          viewBox="0 0 28 28"
          className={'hw-radio-image ' + styles.radio_image}
          focusable="false"
          style={forPrint ? { height: '1em', width: '1em' } : null}
        >
          <circle
            className={`hw-radio-outer-circle ${styles.radio_outer_circle}`}
            cx="14"
            cy="14"
            r="9"
            style={
              forPrint ? { stroke: 'black', fill: 'white', height: '28px', width: '28px' } : null
            }
          />
          <circle
            className={`hw-radio-unchecked ${styles.radio_inner_circle} ${styles.radio_unchecked}`}
            cx="14"
            cy="14"
            r="5"
            style={forPrint ? { display: 'none' } : null}
          />
          <circle
            className={`hw-radio-checked ${styles.radio_inner_circle} ${styles.radio_checked}`}
            cx="14"
            cy="14"
            r="5"
            style={forPrint ? { display: 'none' } : null}
          />
        </svg>
      </span>
    )
  }
}

Radio.propTypes = {
  id: PropTypes.string,
  isRequired: PropTypes.bool,
  name: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
  onClick: PropTypes.func,
  readonly: PropTypes.bool,
  forPrint: PropTypes.bool,
}

Radio.defaultProps = {
  isRequired: false,
  readonly: false,
  forPrint: false,
}

export default Radio
