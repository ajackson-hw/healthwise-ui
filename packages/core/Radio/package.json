{
  "name": "@healthwise/radio",
  "version": "2.0.0",
  "author": "Healthwise",
  "description": "Healthwise UI Assets - radio assets",
  "license": "MIT",
  "homepage": "https://github.com/healthwise/healthwise-ui#readme",
  "main": "index.js",
  "publishConfig": {
    "access": "public"
  },
  "dependencies": {
    "prop-types": "^15.5.7"
  },
  "peerDependencies": {
    "@healthwise/theme": "^4.0.0",
    "react": ">=15.4.2"
  }
}
