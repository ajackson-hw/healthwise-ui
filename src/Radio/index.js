import React, { Component } from 'react'
import PropTypes from 'prop-types'
import styled, { withTheme } from 'styled-components'

import { defaultTheme } from '../Theme'

const Root = styled.span`
  position: relative;
  display: block;

  opacity: ${props => (props.readonly || props.disabled ? 0.35 : 1)};
  cursor: ${props => (props.readonly || props.disabled ? 'not-allowed' : 'default')};
`

const Label = styled.label`
  display: inline-flex;
  align-items: center;
  line-height: 1;
`

const Svg = styled.svg`
  width: ${props => (props.forPrint ? '1em' : '28px')};
  height: ${props => (props.forPrint ? '1em' : '28px')};
`

const OuterCircle = styled.circle`
  width: ${props => (props.forPrint ? '28px' : 'auto')};
  height: ${props => (props.forPrint ? '28px' : 'auto')};
  stroke: ${props => (props.forPrint ? 'black' : props.theme.colorTextPrimary)};
  fill: ${props => (props.forPrint ? 'white' : 'transparent')};
  stroke-width: 2;
`

const InnerCircle = styled.circle`
  stroke: ${props => props.theme.colorTextPrimary};
  fill: ${props => props.theme.colorTextPrimary};
  stroke-width: 1;
  display: none;
`

const Input = styled.input`
  position: absolute;
  display: ${props => (props.forPrint ? 'none' : 'block')};
  height: 28px;
  width: 28px;
  margin: 0;
  padding: 0;
  border: 0;
  outline: 0;
  opacity: 0;
  cursor: pointer;

  :focus + ${Svg} {
    outline: ${props => props.theme.focusIndicator};
    outline-offset: ${props => props.theme.focusIndicatorOffset};
  }

  :focus:invalid {
    border-color: ${props => props.theme.colorError};
  }

  :checked + ${Svg} ${InnerCircle} {
    display: inline-block;
  }

  :disabled {
    cursor: not-allowed;
  }
`

const LabelContent = styled.span`
  padding-left: 0.3rem;
`

class Radio extends Component {
  render() {
    let {
      id,
      name,
      label,
      value,
      isRequired,
      onClick,
      readonly,
      disabled,
      forPrint,
      theme,
      ...otherProps
    } = this.props

    return (
      <Root disabled={readonly || disabled} className="hw-radio-wrapper">
        <Label className="hw-radio-wrapper-label">
          <Input
            type="radio"
            id={id}
            name={name}
            value={value}
            required={isRequired}
            className="hw-radio"
            onClick={onClick}
            disabled={readonly || disabled}
            forPrint={forPrint}
            theme={theme}
            {...otherProps}
          />
          <Svg
            role="presentation"
            viewBox="0 0 28 28"
            className="hw-radio-image"
            focusable="false"
            forPrint={forPrint}
          >
            <OuterCircle
              className="hw-radio-outer-circle"
              cx="14"
              cy="14"
              r="9"
              forPrint={forPrint}
              theme={theme}
            />
            <InnerCircle className="hw-radio-checked" cx="14" cy="14" r="5" theme={theme} />
          </Svg>

          <LabelContent className="hw-checkbox-wrapper-label-text">{label}</LabelContent>
        </Label>
      </Root>
    )
  }
}

Radio.propTypes = {
  id: PropTypes.string,
  isRequired: PropTypes.bool,
  name: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
  onClick: PropTypes.func,
  readonly: PropTypes.bool,
  disabled: PropTypes.bool,
  forPrint: PropTypes.bool,
  theme: PropTypes.shape({
    colorTextPrimary: PropTypes.string,
    colorError: PropTypes.string,
    focusIndicator: PropTypes.string,
    focusIndicatorOffset: PropTypes.string,
  }),
}

Radio.defaultProps = {
  isRequired: false,
  readonly: false,
  forPrint: false,
  theme: defaultTheme,
}

export default withTheme(Radio)
