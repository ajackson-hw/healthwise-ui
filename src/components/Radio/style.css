.radio_wrapper {
  position: relative;
  display: block;
  line-height: 28px;
}

.radio {
  position: absolute;
  display: block;
  height: 28px;
  width: 28px;
  margin: 0;
  padding: 0;
  border: 0;
  outline: 0;
  opacity: 0;
  cursor: pointer;
}

.radio:disabled {
  cursor: default;
}

.radio + .radio_image .radio_outer_circle {
  stroke: var(--color-text-primary);
  fill: transparent;
  stroke-width: 2;
}

.radio:focus + .radio_image .radio_outer_circle {
  stroke: var(--color-text-primary);
}

.radio:focus + .radio_image {
  outline: var(--focus-indicator);
  outline-offset: var(--focus-indicator-offset);
}

.radio + .radio_image {
  width: 28px;
  height: 28px;
  margin-top: 8px;
}

.radio + .radio_image .radio_inner_circle {
  stroke: var(--color-text-primary);
  fill: var(--color-text-primary);
  stroke-width: 1;
  display: none;
}

.radio + .radio_image .radio_unchecked {
  display: none;
}

.radio:checked + .radio_image .radio_unchecked {
  display: none;
}

.radio:checked + .radio_image .radio_checked {
  display: inline-block;
}

.radio:focus + .radio_image .radio_unchecked {
  display: none;
}

.radio:focus:invalid {
  border-color: var(--color-error);
}
