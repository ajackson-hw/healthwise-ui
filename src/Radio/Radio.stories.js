import React from 'react'
import { storiesOf } from '@storybook/react'

import Radio from './index'

const styles = {
  container: {
    fontSize: '24px',
    marginBottom: '16px',
  },
}

storiesOf('core|Form Controls/Radio', module)
  .add(
    'with defaults',
    () => (
      <div>
        <div style={styles.container}>
          <Radio name="test" value="test" />
          <Radio name="test" value="test 2" defaultChecked />
          <Radio name="test" value="test 3" label="This is a label" />
        </div>
      </div>
    ),
    {
      info: `Demonstrates default rendering of Radio component`,
    }
  )
  .add(
    'disabled',
    () => (
      <div>
        <div style={styles.container}>
          <Radio name="test" value="test" disabled />
          <Radio name="test" value="test 2" disabled defaultChecked />
          <Radio name="test" value="test 3" label="This is a label" disabled />
        </div>
      </div>
    ),
    {
      info: `Demonstrates a disabled Radio component`,
    }
  )
