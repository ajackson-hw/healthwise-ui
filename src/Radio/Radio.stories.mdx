import { Meta, Story, Preview } from '@storybook/addon-docs/blocks'
import { action } from '@storybook/addon-actions'

import Radio from './index'

<Meta title="Form Controls/Radio" component={Radio} />

# Radio

Demonstrates default rendering of Radio component.

<Preview>
  <Story name="default">
    <div>
      <Radio name="test" value="test" />
      <Radio name="test" value="test 2" defaultChecked />
      <Radio name="test" value="test 3" label="This is a label" />
    </div>
  </Story>
</Preview>

Demonstrates a disabled Radio component.

<Preview>
  <Story name="disabled">
    <div>
      <Radio name="test" value="test" disabled />
      <Radio name="test" value="test 2" disabled defaultChecked />
      <Radio name="test" value="test 3" label="This is a label" disabled />
    </div>
  </Story>
</Preview>
