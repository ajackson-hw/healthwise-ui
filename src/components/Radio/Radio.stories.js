import React from 'react'
import { storiesOf } from '@storybook/react'

import Radio from './index'

const styles = {
  container: {
    fontSize: '24px',
    marginBottom: '16px',
  },
}

storiesOf('Form Controls/Radio', module).addWithInfo(
  'with defaults',
  `Demonstrates default rendering of Radio component`,
  () => (
    <div>
      <div style={styles.container}>
        <Radio name="test" value="test" />
      </div>
      <div style={styles.container}>
        <Radio name="test 2" value="test" defaultChecked />
      </div>
    </div>
  )
)
