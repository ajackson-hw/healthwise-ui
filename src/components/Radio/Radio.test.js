/* global describe, it, expect, jest */
import React from 'react'
import renderer from 'react-test-renderer'
import Radio from './index'

describe('<Radio />', () => {
  it('renders correctly with minimum props', () => {
    const component = renderer.create(<Radio name="steve" value="awesome" />)
    expect(component).toMatchSnapshot()
  })

  it('uses unused attributes to the input field', () => {
    const component = renderer.create(<Radio name="steve" value="okay" id="awesome" />)
    expect(component).toMatchSnapshot()
  })
})
