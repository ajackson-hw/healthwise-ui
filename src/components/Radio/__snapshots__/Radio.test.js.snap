// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Radio /> renders correctly with minimum props 1`] = `
<span
  className="hw-radio-wrapper radio_wrapper"
>
  <input
    className="hw-radio radio"
    disabled={false}
    id={undefined}
    name="steve"
    onClick={undefined}
    required={false}
    style={null}
    type="radio"
    value="awesome"
  />
  <svg
    className="hw-radio-image radio_image"
    focusable="false"
    role="presentation"
    style={null}
    viewBox="0 0 28 28"
  >
    <circle
      className="hw-radio-outer-circle radio_outer_circle"
      cx="14"
      cy="14"
      r="9"
      style={null}
    />
    <circle
      className="hw-radio-unchecked radio_inner_circle radio_unchecked"
      cx="14"
      cy="14"
      r="5"
      style={null}
    />
    <circle
      className="hw-radio-checked radio_inner_circle radio_checked"
      cx="14"
      cy="14"
      r="5"
      style={null}
    />
  </svg>
</span>
`;

exports[`<Radio /> uses unused attributes to the input field 1`] = `
<span
  className="hw-radio-wrapper radio_wrapper"
>
  <input
    className="hw-radio radio"
    disabled={false}
    id="awesome"
    name="steve"
    onClick={undefined}
    required={false}
    style={null}
    type="radio"
    value="okay"
  />
  <svg
    className="hw-radio-image radio_image"
    focusable="false"
    role="presentation"
    style={null}
    viewBox="0 0 28 28"
  >
    <circle
      className="hw-radio-outer-circle radio_outer_circle"
      cx="14"
      cy="14"
      r="9"
      style={null}
    />
    <circle
      className="hw-radio-unchecked radio_inner_circle radio_unchecked"
      cx="14"
      cy="14"
      r="5"
      style={null}
    />
    <circle
      className="hw-radio-checked radio_inner_circle radio_checked"
      cx="14"
      cy="14"
      r="5"
      style={null}
    />
  </svg>
</span>
`;
